android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/VPNServiceBinder.kt

```diff
@@ -56,7 +56,11 @@ class VPNServiceBinder(service: VPNService) : Binder() {
                     // [data] is here a json containing the wireguard conf
                     val buffer = data.createByteArray()
                     val json = buffer?.let { String(it) }
-                    val config = JSONObject(json)
+                    val config = json?.let { JSONObject(it) }
+                    if (config == null) {
+                        Log.e(tag, "Client provided config was not parsable")
+                        return true
```

Comment:
How come we return true here if there was an error?

Reply:
Ha, it's documented: `returns true if the [code] was accepted` - the function code is accepted, just the payload is wront :)

Reply:
Yeah I read that, but  I assumed if there was an error then the code had not been accepted... Clearly I don't know what is happening here :sos: